src/Aacotroneo/Saml2/Saml2Auth.php

```diff
@@ -59,11 +60,11 @@ function login($returnTo = null)
      * Initiate a saml2 logout flow. It will close session on all other SSO services. You should close
      * local session if applicable.
      */
-    function logout()
+    function logout($returnTo = null)
     {
         $auth = $this->auth;
-
-        $auth->logout();
+        $federationData = Session::get('federationData');
+        $auth->logout($returnTo, array(), $federationData['nameId'], $federationData['sessionIndex']);
```

Comment:
if we could send the federation data as a parameter... we can make them null by default too, and we are done I think